src/architecture/utilities/avsEigenSupport.h

```diff
@@ -64,4 +66,235 @@ Eigen::Vector3d eigenMRPd2Vector3d(Eigen::MRPd vec);
+in order to save an unnecessary conversion between types.
+@return Eigen::Vector
+@param inArray The input array (row-major)
+*/
+template <typename ScalarT, std::size_t size>
+Eigen::Vector<ScalarT, size> cArrayAsEigenVector(ScalarT (&inArray)[size]) {
+    return Eigen::Map<Eigen::Vector<ScalarT, size>>(inArray);
+}
+
+/*! This function performs the conversion between an input C array and an output Eigen vector3. This function is provided
+in order to save an unnecessary conversion between types.
+@return Eigen::Vector
+@param inArray The input array (row-major)
+*/
+template <typename ScalarT>
+Eigen::Vector3<ScalarT> cArrayAsEigenVector3(ScalarT *inArray) {
+    return Eigen::Map<Eigen::Vector3<ScalarT>>(inArray);
+}
+
+/*! This function performs the conversion between an input C array
+and an output Eigen MRP. This function is provided
+in order to save an unnecessary conversion between types.
+@return Eigen::MRP
+@param inArray The input array (row-major)
+*/
+template <typename ScalarT>
+Eigen::MRP<ScalarT> cArrayAsEigenMrp(ScalarT *inArray) {
+    Eigen::MRP<ScalarT> sigma_Eigen;
+    sigma_Eigen = Eigen::Map<Eigen::Vector<ScalarT, 3>>(inArray);
+
+    return sigma_Eigen;
+}
+
+/*! This function performs the conversion between an input C array
+and an Eigen 3x3 matrix. This function is provided
+in order to save an unnecessary conversion between types.
+@return Eigen::Matrix3
+@param inArray The input array (row-major)
+*/
+template <typename ScalarT>
+Eigen::Matrix3<ScalarT> cArrayAsEigenMatrix3(ScalarT *inArray) {
+    return Eigen::Map<Eigen::Matrix3<ScalarT>>(inArray, 3, 3).transpose();
+}
+
+/*! This function performs the conversion between an input C 3x3
+2D-array and an output Eigen Matrix3. This function is provided
+in order to save an unnecessary conversion between types
+@return Eigen::Matrix3
+@param in2DArray The input 2D-array
+*/
+template <typename ScalarT>
+Eigen::Matrix3<ScalarT> c2DArrayAsEigenMatrix3(ScalarT in2DArray[3][3]) {
+    Eigen::Matrix3<ScalarT> outMat;
+    for (int i = 0; i < 3; i++) {
+        for (int j = 0; j < 3; j++) {
+            outMat(i, j) = in2DArray[i][j];
+        }
+    }
+
+    return outMat;
+}
+
+/*! This function converts the Eigen MRP to Vector3
+ @return Eigen::Vector3
+ @param mrp The input Vector3 variable
+ */
+template <typename ScalarT>
+Eigen::Vector3<ScalarT> eigenMrpToVector3(const Eigen::MRP<ScalarT> mrp) {
+    Eigen::Vector3<ScalarT> vec3;
+
+    vec3[0] = mrp.x();
+    vec3[1] = mrp.y();
+    vec3[2] = mrp.z();
+
+    return vec3;
+}
+
+/*! This function returns the Eigen DCM that corresponds to a 1-axis rotation
+ by the angle theta.  The DCM is the positive theta rotation from the original
+ frame to the final frame.
+ @return Eigen::Matrix3
+ @param angle The input rotation angle
+ */
+template <typename ScalarT>
+Eigen::Matrix3<ScalarT> eigenM1(const ScalarT angle) {
+    Eigen::Matrix3<ScalarT> mOut;
```

Comment:
:large_orange_diamond: This and the next three can become `Eigen::Matrix3<ScalarT> mOut = Eigen::Matrix3<ScalarT>::Identity();`